pkg/reconciler/buildtemplate/buildtemplate.go

```diff
@@ -193,3 +182,18 @@ func allCached(desired []caching.Image, observed []*caching.Image) (missing []ca
 	}
 	return missing
 }
+
+func CreateMissingImageCaches(ctx context.Context, client cachingclientset.Interface, missing []caching.Image) error {
```

Comment:
Could this just transparently calculate `missingImageCaches` and create them? Users shouldn't need to call both methods.